tools/StaticAnalysis/Exceptions/Az.HDInsight/SignatureIssues.csv

```diff
@@ -68,3 +68,8 @@
 "Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll","Microsoft.Azure.Commands.HDInsight.ManagementCommands.AddAzureHDInsightClusterIdentity","Add-AzHDInsightClusterIdentity","1","8420","Parameter set 'CertificateFileContents' of cmdlet 'Add-AzHDInsightClusterIdentity' contains at least one parameter with a position larger than four, which is discouraged.","Limit the number of positional parameters in a single parameter set to four or fewer."
 "Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll","Microsoft.Azure.Commands.HDInsight.NewAzureHDInsightClusterCommand","New-AzHDInsightCluster","1","8700","Parameter set '__AllParameterSets', 'CertificateFilePath', 'CertificateFileContents' of cmdlet 'New-AzHDInsightCluster' have the same mandatory parameters, and both of them are not default parameter set which may cause confusion.","Merge these parameter sets into one parameter set."
+"Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll","Microsoft.Azure.Commands.HDInsight.NewAzureHDInsightClusterAutoscaleScheduleConditionCommand","New-AzHDInsightClusterAutoscaleScheduleCondition","1","8100","New-AzHDInsightClusterAutoscaleScheduleCondition Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
+"Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll","Microsoft.Azure.Commands.HDInsight.NewAzureHDInsightClusterAutoscaleScheduleConditionCommand","New-AzHDInsightClusterAutoscaleScheduleCondition","1","8410","Parameter Days of cmdlet New-AzHDInsightClusterAutoscaleScheduleCondition does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
+"Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll","Microsoft.Azure.Commands.HDInsight.NewAzureHDInsightClusterAutoscaleConfiguration","New-AzHDInsightClusterAutoscaleConfiguration","1","8100","New-AzHDInsightClusterAutoscaleConfiguration Does not support ShouldProcess but the cmdlet verb New indicates that it should.","Determine if the cmdlet should implement ShouldProcess and if so determine if it should implement Force / ShouldContinue"
+"Microsoft.Azure.PowerShell.Cmdlets.HDInsight.dll","Microsoft.Azure.Commands.HDInsight.SetAzureHDInsightClusterAutoscaleConfigurationCommand","Set-AzHDInsightClusterAutoscaleConfiguration","1","8700","Parameter set 'LoadAutoscaleByResourceIdParameterSet', 'ScheduleAutoscaleByResourceIdParameterSet' of cmdlet 'Set-AzHDInsightClusterAutoscaleConfiguration' have the same mandatory parameters, and both of them are not default parameter set which may cause confusion.","Merge these parameter sets into one parameter set."
```

Comment:
Set-AzHDInsightClusterAutoscaleConfiguration 
- When the customer run `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId` It is ambiguious for Powershell to bind LoadAutoscaleByResourceIdParameterSet or ScheduleAutoscaleByResourceIdParameterSet. please  fix the issue.

Reply:
> Set-AzHDInsightClusterAutoscaleConfiguration
> 
> * When the customer run `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId` It is ambiguious for Powershell to bind LoadAutoscaleByResourceIdParameterSet or ScheduleAutoscaleByResourceIdParameterSet. please  fix the issue.

Hi @msJinLei , Yes, if customer only input `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId`, the cmdlet will fail. But this is we wanted. We have two parameter sets LoadAutoscale and ScheduleAutoscale, we hope customer had better provide enough parameters to distinguish. For some reasons, we can't mark some parameters as mandatory, because we allow customer to set only one parameter for each parameter set.

Reply:
> > Set-AzHDInsightClusterAutoscaleConfiguration
> > 
> > * When the customer run `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId` It is ambiguious for Powershell to bind LoadAutoscaleByResourceIdParameterSet or ScheduleAutoscaleByResourceIdParameterSet. please  fix the issue.
> 
> Hi @msJinLei , Yes, if customer only input `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId`, the cmdlet will fail. But this is we wanted. We have two parameter sets LoadAutoscale and ScheduleAutoscale, we hope customer had better provide enough parameters to distinguish. For some reasons, we can't mark some parameters as mandatory, because we allow customer to set only one parameter for each parameter set.

I think it is by design, PowerShell will tell customer that you don't provide sufficient parameters or some parameters can't exist at the same time.

Reply:
> > > Set-AzHDInsightClusterAutoscaleConfiguration
> > > 
> > > * When the customer run `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId` It is ambiguious for Powershell to bind LoadAutoscaleByResourceIdParameterSet or ScheduleAutoscaleByResourceIdParameterSet. please  fix the issue.
> > 
> > 
> > Hi @msJinLei , Yes, if customer only input `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId`, the cmdlet will fail. But this is we wanted. We have two parameter sets LoadAutoscale and ScheduleAutoscale, we hope customer had better provide enough parameters to distinguish. For some reasons, we can't mark some parameters as mandatory, because we allow customer to set only one parameter for each parameter set.
> 
> I think it is by design, PowerShell will tell customer that you don't provide sufficient parameters or some parameters can't exist at the same time.

We already have the discussion by email and involve Azure Powershell PM. Let's wait for the conclusion.

Reply:
> > > > Set-AzHDInsightClusterAutoscaleConfiguration
> > > > 
> > > > * When the customer run `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId` It is ambiguious for Powershell to bind LoadAutoscaleByResourceIdParameterSet or ScheduleAutoscaleByResourceIdParameterSet. please  fix the issue.
> > > 
> > > 
> > > Hi @msJinLei , Yes, if customer only input `Set-AzHDInsightClusterAutoscaleConfiguration -ResourceId $resourceId`, the cmdlet will fail. But this is we wanted. We have two parameter sets LoadAutoscale and ScheduleAutoscale, we hope customer had better provide enough parameters to distinguish. For some reasons, we can't mark some parameters as mandatory, because we allow customer to set only one parameter for each parameter set.
> > 
> > 
> > I think it is by design, PowerShell will tell customer that you don't provide sufficient parameters or some parameters can't exist at the same time.
> 
> We already have the discussion by email and involve Azure Powershell PM. Let's wait for the conclusion.

Hi @msJinLei , I have added a switch parameter, could you please help review it again?

Reply:
fixed.

---
In reply to: [484649383](https://github.com/Azure/azure-powershell/pull/12858#discussion_r484649383) [](ancestors = 484649383)

Reply:
fixed by adding switch parameter

---
In reply to: [484702851](https://github.com/Azure/azure-powershell/pull/12858#discussion_r484702851) [](ancestors = 484702851)